### PR TITLE
Remove 'synchronize with build' internal functionality

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
We do not want external errors flowing in and us having to manage their state *through the DiagAnalyzerService*.

Note: the external errors still flow in and are managed by the ExternalErrorDiagnosticUpdateSource.  This allows features like allowing the user to suppress build-diagnostics, but can be done without the DiagAnalyzerService being involved.  This pushes us to have entirely separate systems for live diags (DiagAnalyzerService) and external build diagnostics (ExternalErrorDiagnosticUpdateSource).  

In an LSP-pull world, this works nicely.  The build system and Roslyn are already communicating diagnostics with the host and the host owns things like squiggles and task list.  The host also owns dedupe and superseding of live diags over build diags.  

This will also allow us to trim down ExternalErrorDiagnosticUpdateSource and ProjectExternalErrorReporter in the future.  Those systems have a complex eventing based approach that nothing in roslyn currently listens to.  So we can remove those events in teh future, and just use these as a buffer of build-errors for the rare features that need to know this (currently, only the ability to suppress a build-only diagnostic).

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@mavasani ptal, and let me know if my understanding of these capabilities is correct.

File: src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs [hunk 1]
```diff
@@ -127,70 +127,6 @@ public IEnumerable<StateSet> GetOrCreateStateSets(Project project)
...
-            /// <summary>
-            /// Return <see cref="StateSet"/>s that are added as the given <see cref="Project"/>'s AnalyzerReferences.
-            /// This will never create new <see cref="StateSet"/> but will return ones already created.
-            /// </summary>
-            public ImmutableArray<StateSet> CreateBuildOnlyProjectStateSet(Project project)
-            {
-                var projectStateSets = project.SupportsCompilation
-                    ? GetOrUpdateProjectStateSets(project)
-                    : ProjectAnalyzerStateSets.Default;
-                var hostStateSets = GetOrCreateHostStateSets(project, projectStateSets);
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs:139]: aside, i'd love to know what the different state sets all mean.

File: src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs [hunk 2]
```diff
@@ -212,8 +214,6 @@ async ValueTask ClearErrorsCoreAsync(ProjectId projectId, Solution solution, InP
                 ProcessAndRaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear());
             }
 
-            await SetLiveErrorsForProjectAsync(projectId, ImmutableArray<DiagnosticData>.Empty, GetApplicableCancellationToken(state)).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs:215]: this method only existed to forward to the diagservice, and so it got removed.

File: src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs:139]: HostStateSets - state sets corresponding to analyzers coming from installed analyzer VSIXes, which are applicable for any project opened in the VS Session. The most common use case is for our built-in IDE code style analyzers that ship in Features.dll inside Roslyn VSIX installed with VS.

ProjectStateSets - state sets corresponding to analyzers coming from installed analyzer NuGet packages, which are applicable only to the specific project.

**Review by mavasani — APPROVED** [review]

File: src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs [hunk 3]
```diff
@@ -486,35 +486,20 @@ public void AddNewErrors(DocumentId documentId, DiagnosticData diagnostic)
...
         {
-            return SetLiveErrorsForProjectAsync(lastProjectId, state);
+            SetLiveErrorsForProject(lastProjectId, state);
         }
-
-        return default;
     }
 
-    private async ValueTask SetLiveErrorsForProjectAsync(ProjectId projectId, InProgressState state)
+    private static void SetLiveErrorsForProject(ProjectId projectId, InProgressState state)
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs:498]: will do in followup :)

**Pull request merged**